### DISCUSSION TITLE
CubeCell AB01 v1.2 - SX1262 init errors

### DISCUSSION THREAD

**Discussion by d00m178** [jgromes/RadioLib#1259] [General]
hi!

Im checking how to make CubeCell devices work with RadioLib library.
Im using AB01 v1.2  and this example - [SX126x_Receive_Interrupt.ino](https://github.com/jgromes/RadioLib/blob/master/examples/SX126x/SX126x_Receive_Interrupt/SX126x_Receive_Interrupt.ino)

I changed code to use it with Cubecell - commented line 32 and uncommented line [39](https://github.com/jgromes/RadioLib/blob/master/examples/SX126x/SX126x_Receive_Interrupt/SX126x_Receive_Interrupt.ino#L39):

```
// or using CubeCell
SX1262 radio = new Module(RADIOLIB_BUILTIN_MODULE);
```

if I use:

`  int state = radio.begin(915.0, 125.0, 7, 1, 0x12, 10, 8, 0, false);`
or
`  int state = radio.begin(868.0, 125.0, 7, 1, 0x12, 10, 8, 0, false);`

I receive:
```
Copyright @2019-2020 Heltec Automation.All rights reserved.
[SX1262] Initializing ... failed, code -707
```

if I use:

`  int state = radio.begin(868.0, 125.0, 7, 1, 0x12, 10, 8, 1.6, false);`

init error is:
```
Copyright @2019-2020 Heltec Automation.All rights reserved.
[SX1262] Initializing ... failed, code -10
```

if I use default   `int state = radio.begin();`
it seems start to work:
```
[SX1262] Initializing ... success!
[SX1262] Starting to listen ... success!
```
but how to check what the current LoRa settings?

when I run second AB01 device with code examples/SX126x/SX126x_Transmit_Interrupt/SX126x_Transmit_Interrupt.ino and default radio.begin() - these two devices starts communication between each other - I can see in terminaloutput:

`"[SX1262] Sending another packet ... transmission finished!" 
`
in the terminal of sender device and 

```
[SX1262] Received packet!
[SX1262] Data:		Hello World! #15
[SX1262] RSSI:		-110.00 dBm
[SX1262] SNR:		3.75 dB
[SX1262] Frequency error:	-17.44 Hz
```
in the terminal of receiver device.

but how to check this current settings which devices using for communication?
and how to configure my own settings? at least Frequency and SyncWord - I need use CubeCell devices with another LoRaWAN device so Im checking the examples to know how to prepare my own firmware.
any inputs would be useful.
thank you.

**Comment by jgromes** [comment]
> if I use:
>
>  int state = radio.begin(868.0, 125.0, 7, 1, 0x12, 10, 8, 1.6, false);
>
>init error is:
>
>Copyright @2019-2020 Heltec Automation.All rights reserved.
>[SX1262] Initializing ... failed, code -10

Have you investigated this -10 error code? It doesn't seem so, because doing that you will find [this page](https://jgromes.github.io/RadioLib/group__status__codes.html). That will tell you -10 is `RADIOLIB_ERR_INVALID_CODING_RATE` - unsurprising, since you have set the coding rate value to 1, while the accepted range is 5 to 8. The -707 error is also explained on that same page.

> but how to check what the current LoRa settings?

By reading the intro to the example(s), which says:

```
  For default module settings, see the wiki page
  https://github.com/jgromes/RadioLib/wiki/Default-configuration#sx126x---lora-modem
```

Following that link will take you to a wiki page that lists exactly what you're looking for.

**Comment by d00m178** [comment]
> That will tell you -10 is RADIOLIB_ERR_INVALID_CODING_RATE - unsurprising, since you have set the coding rate value to 1, while the accepted range is 5 to 8

thank you!
but how to find the full list of coding rates values? which rates means "5-8" values?
I set up 1 because I took this value from Heltec examples:
```
#define LORA_CODINGRATE                             1         // [1: 4/5,
                                                              //  2: 4/6,
                                                              //  3: 4/7,
                                                              //  4: 4/8]
```
and I didnt realize that for RadioLib library this values maybe different..

UPD:
ok I found this:
https://jgromes.github.io/RadioLib/class_s_x126x.html#afd3113858966e878e9c67a1e710bd586
and seems 5-8 it is the denominators for 4/*

so if I set 5 it would be 4/5

also - code start to work with
`  int state = radio.begin(868.0, 125.0, 7, 5, 0x34, 10, 8, 1.6, false);
`
Im still not sure if "1.6" is correct value for CubeCell but at least AB01 print output:
`[SX1262] Sending another packet ... transmission finished!
`

the only issue so far - I can not receive this on my receiver (RPI4 + Waveshare LoRaWAN SX1262 HAT) side but I will create new post about this.
thank you again for you work on this library.
